SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java

```diff
@@ -19,11 +23,26 @@ public ConferenceServiceImpl(ConferenceRepository conferenceRepository) {
 
     @Override
     public List<Conference> listConferences() {
-        return conferenceRepository.findAll();
+        List<Conference> conferences = conferenceRepository.findAll();
+        return conferences;
     }
 
     @Override
-    public Optional<Conference> findConferenceById(Long conferenceId) {
-        return conferenceRepository.findById(conferenceId);
+    public Conference findConferenceById(Long conferenceId) throws ConferenceNotFoundException {
+        Optional<Conference> optionalConference = conferenceRepository.findById(conferenceId);
+            return optionalConference.orElseThrow(() -> new ConferenceNotFoundException());
+
+    }
+    @Override
+    public Conference saveConference(Conference conference) throws ConferenceConflictException, InvalidConferenceFieldException {
+        if(conference.getId() == null || conference.getName() == null || conference.getDateTime() == null || conference.getCity() == null || conference.getDescription() == null || conference.getTopic() == null){
+            throw new InvalidConferenceFieldException();
+        }
+        if(conferenceRepository.existsById(conference.getId())) {
+            throw new ConferenceConflictException();
+        } else if(!conference.getDateTime().isAfter(Instant.now())){
+            throw new InvalidConferenceFieldException(); //fails other tests!!!
+        }
```

Comment:
Fix them!

Reply:
Sorry, disregard that I forgot to delete the comment, everything works fine now